### PR TITLE
Add list filtering from channels benchmark

### PR DESCRIPTION
**Pull request by Herklos**
# Lists filtering
## Description
We wanted to test how to perform the better lists filtering.

## Results
With should prefer
```
def comprehension_filtering_with_iter(objects_to_filter, testing_filters):
    return [
            candidate
            for candidate in objects_to_filter
            if all(
                k in candidate
                and (v == WILDCARD or candidate[k] in [v, WILDCARD])
                for k, v in testing_filters.items()
            )
        ]
```

Timy executed (filtering_by_filter) for 1000 times in 0.016979 seconds

Timy executed (filtering_by_filter_with_iter) for 1000 times in 0.017567 seconds

Timy executed (comprehension_filtering) for 1000 times in 0.009410 seconds

Timy executed (comprehension_filtering_with_iter) for 1000 times in 0.009953 seconds

### CODE REVIEW

**Review by GuillaumeDSM — APPROVED** [review]
great !

File: list_operations/filtering/lists_filtering.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+def comprehension_filtering(objects_to_filter, testing_filters):
+    return [ # from Channels._check_filters
+        candidate
+        for candidate in objects_to_filter
+        if all(
+            [
+                k in candidate
+                and (v == WILDCARD or candidate[k] in [v, WILDCARD])
+                for k, v in testing_filters.items()
+            ]
```
Comment (GuillaumeDSM) [inline list_operations/filtering/lists_filtering.py:38]: is it possible not to create this list though (use a generator)

Comment (GuillaumeDSM) [reply list_operations/filtering/lists_filtering.py:38]:
```
all(
            
                k in candidate
                and (v == WILDCARD or candidate[k] in [v, WILDCARD])
                for k, v in testing_filters.items()
)
```

Comment (Herklos) [reply list_operations/filtering/lists_filtering.py:38]: Good idea, I'll check that

**Pull request merged**